helix-term/src/ui/editor.rs

```diff
@@ -999,6 +1033,12 @@ impl Component for EditorView {
                                     consumed = true;
 
                                     if callback.is_some() {
+                                        if let Some(compl) = cx.editor.last_completion.take() {
+                                            self.last_insert
+                                                .1
+                                                .push(InsertEvent::CompletionApply(compl));
+                                        }
```

Comment:
If the event was consumed (even to just close the prompt), then no command was ran == no text edits. Remove this and keep the one below on line 1050